src/lib/battery/battery.cpp

```diff
@@ -307,20 +307,25 @@ float Battery::computeRemainingTime(float current_a)
 	float time_remaining_s = NAN;
 
 	if (_vehicle_status_sub.updated()) {
-		vehicle_status_s vehicle_status;
-
-		if (_vehicle_status_sub.copy(&vehicle_status)) {
-			_armed = (vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_ARMED);
+		if (_vehicle_status_sub.update()) {
```

Comment:
Why is this change needed? Looks out of context to me.

Reply:
I need the vehicle_status later on since i need to check if it is a Fixedwing on the current estimation

Reply:
I reworked it a bit its closer again, but i need the vehicle_status outside that loop